components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -269,6 +269,34 @@ pub fn strcmp(left: &Option<Bytes>, right: &Option<Bytes>) -> Result<Option<i64>
     })
 }
 
+#[rpn_fn]
+#[inline]
+pub fn un_hex(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(arg.as_ref().and_then(|bytes| {
+        let mut v = Vec::with_capacity(bytes.len() / 2);
+        let skip = if bytes.len() % 2 == 1 {
+            // if the length is not the multiple of 2, first value can ignore upper byte with no effect
+            v.push(val(bytes[0])?);
+            1
+        } else {
+            0
+        };
+        for pair in bytes[skip..].chunks(2) {
+            v.push(val(pair[0])? << 4 | val(pair[1])?);
```

Comment:
Could we use `from_hex` instead?

Reply:
We can use [`decode_to_slice`](https://docs.rs/hex/0.4.0/hex/fn.decode_to_slice.html) here.